### ISSUE TITLE
setLocalDescription() | error: +18ms Failed to set local answer sdp: Failed to push down transport description: Offerer must use actpass value for setup attribute.

### ISSUE THREAD

**Issue by gudge25** [versatica/JsSIP#311]
After some period of talking  i got :

jssip.js:21391 JsSIP:RTCSession receiveRequest() +69ms
jssip.js:21391 JsSIP:RTCSession receiveReinvite() +33ms
jssip.js:21391 rtcninja:RTCPeerConnection setRemoteDescription() +79ms
jssip.js:21391 rtcninja:RTCPeerConnection setRemoteDescription() | success +56ms
jssip.js:21391 JsSIP:RTCSession createLocalDescription() +8ms
jssip.js:21391 rtcninja:RTCPeerConnection createAnswer() +8ms
jssip.js:21391 rtcninja:RTCPeerConnection onsignalingstatechange() | signalingState: have-remote-offer +15ms
jssip.js:21391 rtcninja:RTCPeerConnection createAnswer() | success +20ms
jssip.js:21391 rtcninja:RTCPeerConnection setLocalDescription() +7ms
jssip.js:21566 rtcninja:ERROR:RTCPeerConnection setLocalDescription() | error: +18ms Failed to set local answer sdp: Failed to push down transport description: Offerer must use actpass value for setup attribute.
jssip.js:21391 JsSIP:Transport sending WebSocket message:

SIP/2.0 500 JsSIP Internal Error
Via: SIP/2.0/WS 77.91.132.8:5060;branch=z9hG4bK3d86bf97;rport
To: <sip:1060@77.91.132.8>;tag=jgpreu0pf5
From: <sip:s@77.91.132.8>;tag=as0236cdf8
Call-ID: 6alkdf598o5maeejv1ud
CSeq: 102 INVITE
Supported: timer,ice,outbound
Content-Length: 0

Thanks for help

**Comment by gudge25** [comment]
is usefull
this is a sip log 
http://pastebin.com/bHAfWiQT

**Comment by ibc** [comment]
This is not a bug in JsSIP. The SDP offer must include "a=setup:actpass" as per spec, but your has "a=setup:active". That's an Asterisk issue.

**Comment by gudge25** [comment]
not sure because when i use this demo for test http://sipjs.com/demo-phone/
Everything is ok

**Comment by gudge25** [comment]
there might be some different probled

**Comment by jmillan** [comment]
Hi @gudge25,

Yes, your server is violating a MUST in RFC 5763, section 5: 
```
      The endpoint that is the offerer MUST use the setup attribute
      value of setup:actpass and be prepared to receive a client_hello
      before it receives the answer.
```

On the other hand, Chrome is not able to change the DTLS role once connection is established:

`https://code.google.com/p/webrtc/issues/detail?id=2782`

Which tell me that if this ever works somewhere is because on the browser side, the `a=setup` line is mangled to coincide with the corresponding peer's DTLS role; and since ICE re-negotiation is not done on SIP re-negotiation, then the SDP DTLS attributes are simply ignored.

In summary: You can mangle the SDP `a=setup` attribute on incoming SIP re-negotiations to make it match the initial peer's role.

**Comment by fukemy** [comment]
i have a=setup:active but still got this error
